spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/MessageChannelToInputFluxParameterAdapter.java

```diff
@@ -71,7 +73,9 @@ public boolean supports(Class<?> boundElementType, MethodParameter methodParamet
 			return Flux.create(emitter -> {
 				MessageHandler messageHandler = message -> {
 					synchronized (monitor) {
-						if (argumentClass.isAssignableFrom(message.getPayload().getClass())) {
+						if (argumentClass.equals(Object.class)
```

Comment:
`argumentClass` may not always be `Object.class`, but it seems like this condition is already covered by the second part of the expression (i.e. if true, `argumentClass` it would always be assignable from anything)